content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+
+
+## Making multiple tests
+
+The natural next step is to run multiple tests for all potential risk factors
+recorded. There are many ways to go about this in **R**, but we will illustrate
+one of the most common workflows:
+
+1. isolate a subset of variables to test (the ones indicative of food consumption) (using `[]`)
+2. for each:
+    i) build contingency table with illness status (using `table`)
+    ii) run a Chi-square test on this table (using `chisq.test`)
+
+All of these have been seen so far; the only missing piece is step 2), which we
+will cover using a very handy function called `lapply()`.
+
+
+### Isolating the variables to test
+
+We use the natural structure of the dataset, as variables representing food
+consumption are all the last columns:
+
+```{r stegen_isolate_food}
+names(stegen)
+```
+
+In this case, we need to retain columns 6 to 21, excluding `tportion` and
+`mportion`, which are not binary; this can be done using the subsetting operator
+`[...]`, where `...` are numbers indicating the columns to retain:
+
+```{r stegen_food}
+to_keep <- c(6, 8, 9, 10, 12:21)
+to_keep
+food <- stegen[to_keep]
+food
+```
+
+
+### Building several contingency tables
+
+There are many variables to test, and having to enter separate command lines for
+each would be cumbersome and prone to errors. As a workaround, the function
+`lapply()` can be used. This function allows to repeat an operation using a vector
+of inputs, using each element in turn. Its general syntax is
+`lapply(vector_of_inputs, function_to_use, other_arguments)`, where
+`other_arguments` can be any secondary arguments taken by
+`function_to_use()`. Here, we use it to build a separate contingency table of each
+variable in `food` crossed with illness status (`stegen$ill`); note that we show
+only the first few tables:
+
+```{r stegen_food_tables, echo = c(1,3), eval = -2}
+food_tables <- lapply(food, table, stegen$ill)
+food_tables
+head(food_tables)
+```
+
+
+### Realising multiple Chi-squared tests
+
+The same principle used to compute several contingency tables can be used to
+compute Chi-square tests for each table, using `lapply()`; note that we show
+only the first few tests:
+
+```{r stegen_food_tests, echo = c(1,3), eval = -2}
+food_tests <- lapply(food_tables, chisq.test)
+food_tests
+head(food_tests)
+```
+
+Looking at these results, it seems a large number of food items are
+significantly correlated to the illness. Which food item is the biggest suspect?
+We will try and address this question using risk ratios.
+
+
+
+## Risk ratios
+
+### For one variable 
+
+The [risk ratio](https://en.wikipedia.org/wiki/Risk_ratio) is defined as the
+ratio between the proportion of illness in one group (typically 'exposed') vs
+another ('non-exposed'). For instance, let us consider the contingency table of
+pork consumption and illness:
+
+```{r stegen_pork_table}
+food_tables$pork
```

Comment:
If we hadnt previously explained lists - then this might be confusing, "why does the dollar sign before only give one variable and here it gives a whole dataframe?" type question...